jupyterlab/extensions/pypi.py

```diff
@@ -52,8 +52,10 @@ def make_connection(self, host):
 xmlrpc_transport_override = None
 
 all_proxy_url = environ.get("ALL_PROXY")
-http_proxy_url = environ.get("HTTP_PROXY") or all_proxy_url
-https_proxy_url = environ.get("HTTPS_PROXY") or all_proxy_url or http_proxy_url
+http_proxy_url = environ.get("http_proxy") or environ.get("HTTP_PROXY") or all_proxy_url
```

Comment:
```suggestion
# For historical reasons, we also support the lowercase environment variables.
# Info: https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/
http_proxy_url = environ.get("http_proxy") or environ.get("HTTP_PROXY") or all_proxy_url
```

To avoid a well-intentioned developer ignoring or removing `http_proxy` and `https_proxy` in the future, it might be good to include a comment. Prior to this issue and PR, I hadn't seen a lowercase environment variable on a UNIX-like system.

Reply:
It is a weird, ancient exception to the uppercase env vars custom, but one that is deeply embedded in some core tools and libraries like curl.